vllm/entrypoints/openai/serving_engine.py

```diff
@@ -175,6 +177,10 @@ def _validate_prompt_and_tokenize(
             prompt).input_ids
         token_num = len(input_ids)
 
+        # Note: EmbeddingRequest doesn't have max_tokens
```

Comment:
I disagree. There is a property in config called `max_position_embeddings`:
- https://huggingface.co/intfloat/e5-mistral-7b-instruct/blob/main/config.json#L12
- https://huggingface.co/intfloat/e5-base-v2/blob/main/config.json#L15

And it's normally something rather small. `e5-mixtral-7b-instruct` is a bit unique with max context 32k :)

Reply:
The [OpenAI official API](https://platform.openai.com/docs/api-reference/embeddings) doesn't include max_tokens in the embedding request.

Reply:
So that means any old (embeddings) model can compute embeddings on any length of input without issues?

Reply:
No, it still can't exceed the model's max context length, which the scheduler checks.

Reply:
vLLM checks both fields in model_config: https://github.com/vllm-project/vllm/blob/main/vllm/config.py#L932
So added check for `token_num > self.max_model_len` in `serving_engine`.

Reply:
Yeah, I guess I didn't understand the comment about `max_tokens`. I didn't realize this function also works for non-embedding requests.